data/actions/scripts/other/potions.lua

```diff
@@ -33,126 +79,78 @@ function onUse(player, item, fromPosition, target, toPosition, isHotkey)
-		player:addItem(strongEmptyPot, 1)
-	elseif itemId == strongManaPot then
-		if (not isInArray({1, 2, 3, 5, 6, 7}, target:getVocation():getId()) or target:getLevel() < 50) and not getPlayerFlagValue(player, PlayerFlag_IgnoreSpellCheck) then
-			player:say("This potion can only be consumed by sorcerers, druids and paladins of level 50 or higher.", TALKTYPE_MONSTER_SAY)
-			return true
-		end
-
-		if not doTargetCombatMana(0, target, 115, 185, CONST_ME_MAGIC_BLUE) then
-			return false
-		end
-
-		player:addCondition(exhaust)
-		target:say("Aaaah...", TALKTYPE_MONSTER_SAY)
-		item:remove(1)
-		player:addItem(strongEmptyPot, 1)
-	elseif itemId == greatSpiritPot then
-		if (not isInArray({3, 7}, target:getVocation():getId()) or target:getLevel() < 80) and not getPlayerFlagValue(player, PlayerFlag_IgnoreSpellCheck) then
-			player:say("This potion can only be consumed by paladins of level 80 or higher.", TALKTYPE_MONSTER_SAY)
-			return true
-		end
-
-		if not doTargetCombatHealth(0, target, COMBAT_HEALING, 250, 350, CONST_ME_MAGIC_BLUE) or not doTargetCombatMana(0, target, 100, 200, CONST_ME_MAGIC_BLUE) then
-			return false
-		end
-
-		player:addCondition(exhaust)
-		target:say("Aaaah...", TALKTYPE_MONSTER_SAY)
-		item:remove(1)
-		player:addItem(greatEmptyPot, 1)
-	elseif itemId == greatHealthPot then
-		if (not isInArray({4, 8}, target:getVocation():getId()) or target:getLevel() < 80) and not getPlayerFlagValue(player, PlayerFlag_IgnoreSpellCheck) then
-			player:say("This potion can only be consumed by knights of level 80 or higher.", TALKTYPE_MONSTER_SAY)
-			return true
-		end
-
-		if not doTargetCombatHealth(0, target, COMBAT_HEALING, 425, 575, CONST_ME_MAGIC_BLUE) then
-			return false
+	local stats = potions[potion:getId()]
+	if stats then
+		if (stats.vocations or stats.minLevel) and not player:getGroup():getAccess() then
+			if not stats.minLevel then
+				stats.minLevel = 1
+			end
+			if not stats.vocations then
+				stats.vocations = { [1] = player:getVocation():getId()}
+			end
+
+			if not isInArray(stats.vocations, player:getVocation():getId()) or player:getLevel() < stats.minLevel then
+				local str, vocations = "", {}
+				-- We only show unpromoted vocations in description.
+				for k, v in ipairs(stats.vocations) do
+					if Vocation(v):getDemotion() then
+						table.insert(vocations, v)
+					end
+				end
+				if #vocations > 0 then
+					for i = 1, #vocations do
+						local voc = Vocation(vocations[i])
+						if not voc:getDemotion() then -- Elder druid won't be showed;
+							if i == #vocations then
+								str = str .. (#vocations ~= 1 and " and " or "") .. voc:getName():lower() .. "s"
+							else
+								str = str .. (i ~= 1 and ", " or "") .. voc:getName():lower() .. "s"
+							end
+						end
+					end
+				end
+				vocations = nil
+				if str == "" then
+					str = "players"
+				end
+				player:say("This potion can be consumed by ".. str .. " of level " .. stats.minLevel .. " or higher.", TALKTYPE_MONSTER_SAY)
```

Comment:
I liked your idea, i have made it very possible to change this easily via config.
Check the last 2 commits!